src/cr/cube/matrix.py

```diff
@@ -91,8 +91,32 @@ def blocks(cls, cube_result_matrix, measure_propname=None):
         """
         return cls(cube_result_matrix, measure_propname)._blocks
 
+    @lazyproperty
+    def _base_values(self):
+        """Base values must be implemented by each subclass."""
+        raise NotImplementedError
+
     @lazyproperty
     def _blocks(self):
+        """base, row and col insertion, and intersection matrices."""
+        return [
+            [self._base_values, self._subtotal_columns],
+            [self._subtotal_rows, self._intersections],
+        ]
+
+    @lazyproperty
+    def _intersections(self):
+        """Intersections must be implemented by each subclass."""
+        raise NotImplementedError
+
+    @lazyproperty
+    def _subtotal_columns(self):
+        """Subtotal columns must be implemented by each subclass."""
+        raise NotImplementedError
+
+    @lazyproperty
+    def _subtotal_rows(self):
+        """Subtotal rows must be implemented by each subclass."""
         raise NotImplementedError
```

Comment:
If these will have no eventual implementation, you should probably complete the (non-)implementation now since there won't be a trigger to do it later and "interface-defining" base-class methods like this are not usually tested directly. The best implementation I've found so far is this:

```python
@lazyproperty
def _subtotal_columns(self):
    """ ... """
    raise NotImplementedError(
        "`%s` must implement `._subtotal_columns` % type(self).__name__
    )
```

That way you get an error like "`_SumSubtotals` must implement `._subtotal_columns`" at just the right time (when it's time to implement that) and there's no digging around to work out where it needs implementing. Using `type(self).__name__` doesn't work with old-style Python2 classes (those that don't inherit from `object`) but we shouldn't be encountering any of those. Those would need `self.__class__.__name__` instead.

Reply:
Great how this got resolved by itself when I addressed your previous comments. It's inspiring!